src/components/SocialMediaLinks/SocialMediaLinks.test.tsx

```diff
@@ -0,0 +1,476 @@
+import { render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+import * as React from "react";
+import renderer from "react-test-renderer";
+import SocialMediaLinks from "./SocialMediaLinks";
+import { socialMediaDataMap } from "./SocialMediaDataMap";
+
+// If you want to see what's happening.
+// screen.debug();
+
+describe("SocialMediaLinks Accessibility", () => {
+  describe("Labels", () => {
+    it("passes axe accessibility test without visible labels", async () => {
+      const { container } = render(<SocialMediaLinks />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with labels set to true", async () => {
+      const { container } = render(<SocialMediaLinks showLabels={true} />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+  describe("Colors", () => {
+    it("passes axe accessibility test with link color & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} color={"link"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with textInverse color & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} color={"textInverse"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+  describe("Borders", () => {
+    it("passes axe accessibility test with straight borders & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} border={"straight"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with circular borders", async () => {
+      const { container } = render(<SocialMediaLinks border={"circular"} />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+});
+
+describe("SocialMediaLinks", () => {
+  it("Renders default", () => {
+    render(<SocialMediaLinks />);
+
+    const links = screen.getAllByRole("link");
+
+    // All the links should be there.
+    // @todo I dunno. Isn't this kinda cheating? It will always be true, so why bother?
```

Comment:
It's to build confidence in the default case that we expect certain values to display. You can use a magic number and say "9" instead of `socialMedaDataMap.length`. If the business case updates and we add new logos, then this test will fail and we'll know that something has been updated. That will also help developers keep up-to-date with ux and product updates.

Reply:
Hmm. I guess I am looking for a use case where these tests catch some genuine error rather than fail when there is any change to the expected outcome of the component. For example, adding a new allowed social media type would cause the  count test to fail. But the problem is with the test and not the component. So if we are always updating tests to match expectations, what is the point of the tests?

Reply:
Yes, that's a good point. This is also a special case where the DS component will always render a specific default case whereas other components depend on user input. Is it likely that unit tests and snapshot tests are overkill for this component's default case? Perhaps. Our general testing conventions and patterns don't necessarily fit well with the default case test, but for the non-default case where the dev can pass any data array, it builds confidence on various level of DOM structure, default data rendering, and visual design implementation. 

It seems like there's no point to the test but would appreciate having them for consistency. If, at some point, the default case changes for whatever reason with new updates, the test can help catch any problems.

I'm more concern about testing the cases where the user passes a valid array of social media object and an array of invalid values.